src/gt4py/next/program_processors/runners/dace_fieldview/transformations/auto_optimize.py

```diff
@@ -169,6 +35,7 @@ def gt_auto_optimize(
     reuse_transients: bool = False,
     gpu_launch_bounds: Optional[int | str] = None,
     gpu_launch_factor: Optional[int] = None,
+    constant_symbols: Optional[dict[str, Any]] = None,
```

Comment:
I am actually not sure, but I see your point.
Correct me if I am wrong, but if we would use `typeclass` this would make it impossible to use for example `np.int32`.

Another reason to use `Any` is that at the end we go straight to DaCe's replacement function and according to their signature they want `dict[str, str]`.
However, they perform some conversion step, see `dace/sdfg/replace.py:66`.
Thus I think `Any` is much more appropriate here.